packages/telemetry/src/telemetry.ts

```diff
@@ -289,22 +293,31 @@ class GitProperties {
   }
 
   static async state(cwd?: PathLike): Promise<GitState> {
-    const commandProcess = (await spawn('git', ['status'], {
-      shell: true,
-      cwd: cwd?.toString(),
-    })) as ChildProcess;
-    return new Promise((resolve) => {
-      commandProcess.on('exit', (code) => {
-        switch (code) {
-          case 127:
-            return resolve(GitState.NotInstalled);
-          case 128:
-            return resolve(GitState.NoRepository);
-          default:
-            return resolve(GitState.Ok);
-        }
-      });
-    });
+    const cmd = new CommandStruct(
+      'git',
+      ['status'],
+      cwd?.toString() || '' // perhaps cwd should never be undefined
+    );
+    let returnCode;
+    try {
+      const commandReturn: CommandReturn = await run(cmd);
+      returnCode = commandReturn.code;
+    } catch (err) {
+      if (err instanceof ChildProcessError) {
+        returnCode = err.code;
+      } else {
+        returnCode = GitState.OtherError;
+        return returnCode;
+      }
+    }
+    switch (returnCode) {
+      case 127:
+        return GitState.NotInstalled;
+      case 128:
+        return GitState.NoRepository;
+      default:
+        return GitState.Ok;
+    }
```

Comment:
What about this actually fixed the issue?

Reply:
When I remove the exports commit https://github.com/getappmap/appmap-js/pull/1026/commits/e38fd196d417c75fb980848fc64f623054742827 and add debug statements I see `sendTelemetry` returns without the process exiting.

![rm_exports](https://user-images.githubusercontent.com/111290954/218874695-572e591c-8d7e-4ee7-9c76-18c94a124237.png)